### ISSUE TITLE
ncm-puppet: puppet.pm should add quote to strings in generated YAML file

### ISSUE THREAD

**Issue by Pansanel** [quattor/configuration-modules-core#1374]
Actually, the https://github.com/quattor/configuration-modules-core/blob/master/ncm-puppet/src/main/perl/puppet.pm module generate string without quota in hiera YAML file. It leads to an issue when the string contains a special character, like a comma!

For example, in my JSON profile, I have the following item in the ```hieradata``` list:
```
"gridftp_3a_3aparams_3a_3aglobus_5ftcp_5fport_5frange": "24000,25000"
```

The generated hieradata YAML file contains the following line:
```
gridftp::params::globus_tcp_port_range: 24000,25000
```

When it is used in the Puppet ERB templates, it gives:
```
2400025000
```

However, this syntax isused:
```
gridftp::params::globus_tcp_port_range: '24000,25000'
```

It gives the expected results:
```
24000,25000
```

**Comment by jrha** [comment]
Hmm, strange given this is using `YAML::XS` to do the rendering.
I wonder whether if it is an unintended side-effect of:
https://github.com/quattor/configuration-modules-core/blob/7e0cbf4c9e35560fe84163329ce314f79bc5c77e/ncm-puppet/src/main/perl/puppet.pm#L15

**Comment by sartiran** [comment]
Hi,
indeed it may be a bug in in YAML::XS or most probably a bad choice of the modules conf parameters. I'll make a couple of tests (and see how the puppet ERB templates react to different syntax) and see if I can find a solution to this.

**Comment by jrha** [comment]
Ok, brilliant thanks!

**Comment by Pansanel** [comment]
If you execute this code with ruby
```
my_port = 24000,25000
print my_name.to_s
```
You obtain the following ```[24000, 25000]``` (a list). So, the erb template have the expected behavior.

**Comment by sartiran** [comment]
Hi,

the syntax 

```
prefix '/software/components/puppet/hieradata';

"{gridftp::params::globus_tcp_port_range}" = '22000\,25000';

```

seems to do the job. 
I do not know how to fix the issue otherwise. Or, better said, I'm not sure that this is worth. With YAML::XS there is no way to force the quoting of one string specifically, or event to force quoting of all strings. We can switch to YAML::Syck but, AFAIKS, this module seems to be older and less maintained. Moreover triggering the quoting of all strings it is not backward compatible with the current version of dpm templates using ncm-puppet (e.g. unquoted no/yes are used and are interpreted as booleans). 
What do you think?

Regards,
Andrea

**Comment by jrha** [comment]
I wuld rather not move to a less supported module to solve the issue, maybe it just needs documenting as a known behaviour.

**Comment by jrha** [comment]
Or we write a TT file to produce YAML.

**Comment by Pansanel** [comment]
We should be able to hire a perl guru in our lab next week. He will work on this issue. In the meantime, we may just document the behaviour.

**Comment by jrha** [comment]
I've created an issue for the documentation and moved fixing this to next year.

**Comment by Pansanel** [comment]
Fixed by #1417
